OrbitGl/GraphTrack.cpp

```diff
@@ -0,0 +1,106 @@
+#include "GraphTrack.h"
+
+#include "GlCanvas.h"
+
+GraphTrack::GraphTrack(TimeGraph* time_graph) {
+    SetTimeGraph(time_graph);
+}
+
+void GraphTrack::Draw(GlCanvas* canvas, bool picking) {
+  UNUSED(picking);
+
+  TimeGraphLayout& layout = time_graph_->GetLayout();
+  //float trackHeight = GetHeight();
+  float trackWidth = canvas->GetWorldWidth();
+
+  m_Pos[0] = canvas->GetWorldTopLeftX();
+  SetSize(trackWidth, GetHeight());
+
+  float x0 = m_Pos[0];
+  float x1 = x0 + m_Size[0];
+
+  float y0 = m_Pos[1];
+  float y1 = y0 - m_Size[1];
+
+  if (m_Picked) {
+    glColor4ub(0, 128, 255, 128);
+  }
+
+  float track_z = layout.GetTrackZ();
+  float text_z = layout.GetTextZ();
+
+  glColor4ubv(&m_Color[0]);
+
+  glBegin(GL_QUADS);
+  glVertex3f(x0, y0, track_z);
+  glVertex3f(x1, y0, track_z);
+  glVertex3f(x1, y1, track_z);
+  glVertex3f(x0, y1, track_z);
+  glEnd();
+
+  if (canvas->GetPickingManager().GetPicked() == this)
+    glColor4ub(255, 255, 255, 255);
+
+  glBegin(GL_LINES);
+  glVertex3f(x0, y0, track_z);
+  glVertex3f(x1, y0, track_z);
+  glVertex3f(x1, y1, track_z);
+  glVertex3f(x0, y1, track_z);
+  glEnd();
+
+  Color col(0, 128, 255, 128);
+  Color colors[2];
+  Fill(colors, col);
+
+  // Current time window
+  uint64_t min_ns = time_graph_->GetTickFromUs(time_graph_->GetMinTimeUs());
+  uint64_t max_ns = time_graph_->GetTickFromUs(time_graph_->GetMaxTimeUs());
+  double time_range = static_cast<float>(max_ns - min_ns);
+  if (values_.size() < 2 || time_range == 0) return;
+
+  auto it = values_.lower_bound(min_ns);
+  if (it == values_.end()) return;
+  uint64_t last_time = it->first;
```

Comment:
Using "previous" instead of "last" might make it easier to understand. "Last" could also mean the very last timestamp in the current time window.

Reply:
Done.